plugin/core/windows.py

```diff
@@ -440,24 +444,34 @@ def end_session(self, config_name: str) -> None:
             debug("unloading session", config_name)
             self._sessions[config_name].end()
 
-    def _ensure_project_path(self) -> 'Optional[str]':
-        if self._project_path is None:
-            self._project_path = get_project_path(self._window)
-            if self._project_path is None and self._projectless_root_path is None:
+    def _ensure_workspace(self) -> 'Optional[WorkspaceFolder]':
+        if self._workspace is None:
+            self._workspace = maybe_get_first_workspace_from_window(self._window)
+            if self._workspace is None and self._projectless_workspace is None:
                 # the projectless fallback will only be set once per window.
-                self._projectless_root_path = get_active_view_path(self._window)
-        return self._project_path or self._projectless_root_path
+                view = self._window.active_view()
```

Comment:
I preferred this logic tucked away in `get_active_view_path`, now this function's logic is getting quite nested.